stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -155,10 +155,14 @@ function _chol!(A::AbstractMatrix, ::Type{UpperTriangular})
     n = checksquare(A)
     @inbounds begin
         for k = 1:n
+            Akk = A[k,k]
+            if Akk isa Complex && Akk != Akk'
+                Akk = real(Akk)
+            end
             for i = 1:k - 1
-                A[k,k] -= A[i,k]'A[i,k]
+                Akk -= A[i,k]'A[i,k]
             end
```

Comment:
To be equivalent to the old code, it should set
```jl
A[k,k] = Akk
```
at the end, so that the final `A` has the same values along the diagonal as in previous Julia versions.

Reply:
You are right, that would change the output in case of an error return - will change that.